source/server/http/admin.cc

```diff
@@ -726,6 +725,37 @@ PrometheusStatsFormatter::statsAsPrometheus(const std::vector<Stats::CounterShar
     }
     response.add(fmt::format("{0}{{{1}}} {2}\n", metric_name, tags, gauge->value()));
   }
+
+  for (const auto& histogram : histograms) {
+    const std::string tags = formattedTags(histogram->tags());
+    const std::string metric_name = metricName(histogram->tagExtractedName());
+    if (metric_type_tracker.find(metric_name) == metric_type_tracker.end()) {
+      metric_type_tracker.insert(metric_name);
+      response.add(fmt::format("# TYPE {0} histogram\n", metric_name));
+    }
+
+    const Stats::HistogramStatistics& stats = histogram->cumulativeStatistics();
+    const std::vector<double>& supported_buckets_ref = stats.supportedBuckets();
+    for (size_t i = 0; i < supported_buckets_ref.size(); ++i) {
+      double bucket = supported_buckets_ref[i];
+      double value = stats.computedBuckets()[i];
+      if (histogram->tags().size() > 0) {
```

Comment:
I think instead of this if/else here (and below), it would be easier to read if you had a variable `const std::string hist_tags = histogram->tags().empty() ? EMTPY_STRING : (tags + ",");` Or use absl::StrJoin(). But then only have a single format string.